From 7f4592859b17f8261b32e932bed99733b3250406 Mon Sep 17 00:00:00 2001
From: Christophe Dumez <christophe.dumez@intel.com>
Date: Fri, 10 Jun 2011 15:23:38 +0300
Subject: [PATCH 2/2] Stop using /etc/timezone as fallback for detecting local
 timezone

MeeGo is storing its timezone information in /etc/sysconfig/clock,
not /etc/timezone (unlike Debian-based distributions).

This fallback mecanism is used when Connman.Clock is unavailable,
for example in a chroot environment.
---
 src/kdedate/ksystemtimezone.cpp |   16 ++++++++++------
 1 files changed, 10 insertions(+), 6 deletions(-)

diff --git a/src/kdedate/ksystemtimezone.cpp b/src/kdedate/ksystemtimezone.cpp
index e5901aa..0a6984d 100644
--- a/src/kdedate/ksystemtimezone.cpp
+++ b/src/kdedate/ksystemtimezone.cpp
@@ -299,14 +299,18 @@ void KSystemTimeZonesPrivate::readConfig(bool init)
 #if !defined(QT_NO_DEBUG_OUTPUT)
        kError() << "cannot get localzone from net.connman.Clock";
       // assumption: running connmand is not officially supported in a chroot,
-      //             getting localzone from /etc/timezone
-      kDebug() << "get localzone from /etc/timezone"; 
+      //             getting localzone from /etc/sysconfig/clock
+      kDebug() << "get localzone from /etc/sysconfig/clock";
 #endif
-      QFile f("/etc/timezone");
+      QFile f("/etc/sysconfig/clock");
+      static QRegExp zone_regex("ZONE=\"([^\"]+)\"");
       if (f.open(QIODevice::ReadOnly)) {
-	QTextStream str(&f);
-	m_localZoneName = str.readLine();
-	f.close();
+        QString content = f.readAll();
+        if (zone_regex.indexIn(content) >= 0) {
+          m_localZoneName = zone_regex.cap(1);
+          kDebug() << "localzone from /etc/sysconfig/clock: " << m_localZoneName;
+        }
+        f.close();
       }
     }
 #endif
-- 
1.7.4.4

